lib/chef/provisioning/fog_driver/providers/xenserver.rb

```diff
@@ -0,0 +1,209 @@
+            if server.state == 'Halted'
+              action_handler.perform_action "start machine #{machine_spec.name} (#{server.id} on #{driver_url})" do
+                server.start
+                machine_spec.location['started_at'] = Time.now.to_i
+              end
+              machine_spec.save(action_handler)
+            end
+          end
+
+          def converge_floating_ips(action_handler, machine_spec, machine_options, server)
+            # XenServer does not have floating IPs
+          end
+        end
+      end
+    end
+  end
+end
+
+
+# Add methods required by the fog driver to XenServer's Server class
+require 'fog/compute/models/server'
+module Fog
+  module Compute
+    class XenServer
+      class Server < Fog::Compute::Server
+        def id
+          uuid
+        end
+
+        def state
+          attributes[:power_state]
+        end
+
+        def public_ip_address
+          if xenstore_data['vm-data/ip']
+            xenstore_data['vm-data/ip']
+          else
+            wait_for { tools_installed? }
+            if tools_installed?
+              guest_metrics.networks.first[1]
+            else
+              fail 'Unable to return IP address. Virtual machine does not ' \
+              'have XenTools installed or a timeout occurred.'
+            end
+          end
+        end
+
+        def ready?
+          running?
+        end
+      end
+    end
+  end
+end
+
+#
+# Use call_async instead of call on XMLPRPC::Client
+# Otherwise machine_batch will fail since parallel calls will clash.
+#
+# See http://ruby-doc.org//stdlib-2.1.1//libdoc/xmlrpc/rdoc/XMLRPC/Client.html
+# 
+module Fog
+  module XenServer
+    class Connection
+      require 'xmlrpc/client'
+      attr_reader :credentials
+
+      def request(options, *params)
+        begin
+          parser = options.delete(:parser)
+          method = options.delete(:method)
+
+          if params.empty?
+            response = @factory.call_async(method, @credentials)
+          else
+            if params.length.eql?(1) and params.first.is_a?(Hash)
+              response = @factory.call_async(method, @credentials, params.first)
+            elsif params.length.eql?(2) and params.last.is_a?(Array)
+              response = @factory.call_async(method, @credentials, params.first, params.last)
+            else
+              response = eval("@factory.call_async('#{method}', '#{@credentials}', #{params.map { |p| p.is_a?(String) ? "'#{p}'" : p }.join(',')})")
+            end
+          end
+          raise RequestFailed.new("#{method}: " + response["ErrorDescription"].to_s) unless response["Status"].eql? "Success"
+          if parser
+            parser.parse(response["Value"])
+            response = parser.response
+          end
+
+          response
+        end
+      end
+    end
+
+  end
+end
+
+module Fog
+  class Logger
+    def self.deprecation(message)
+      # Silence...ahh
```

Comment:
Sooooooooo... I understand the desire to do this, but its probably a bad idea in the long run.  Could we perhaps redirect them to `Chef::Log.debug` with a short header?